evm/src/cpu/kernel/asm/hash/blake2b/util.asm

```diff
@@ -0,0 +1,47 @@
+// Load a 64-bit word from kernel general memory.
+%macro mload_blake2b_word_from_bytes
+    // stack: offset
+    DUP1
+    %mload_kernel_general_u32_LE
+    // stack: lo, offset
+    SWAP1
+    // stack: offset, lo
+    %add_const(4)
+    %mload_kernel_general_u32_LE
+    // stack: hi, lo
+    %shl_const(32)
+    // stack: hi << 32, lo
+    OR
+    // stack: (hi << 32) | lo
+%endmacro
+
+// Invert the order of the four bytes in a word.
+%macro invert_four_byte_word
+    // stack: word
+    %mul_const(0x1000000010000000100)
+    %and_const(0xff0000ff00ff00000000ff0000)
+    %mod_const(0xffffffffffff)
+    // stack: word_inverted
+%endmacro
+
+// Invert the order of the eight bytes in a Blake word.
```

Comment:
could then put this in the `basic_macros.asm` file next to `reverse_bytes_u32` and give it a content-agnostic name such as `reverse_bytes_u64`---could in theory also be slightly faster to not separate this into 2 macros but do it directly.